structuredheadings/plugin.js

```diff
@@ -1,35 +1,72 @@
 (function () {
-	CKEDITOR.plugins.add('structuredheadings', {
-		icons: 'structuredheadings',
-		init: (editor) => {
-			// The command to add an autonumbering class
-			editor.addCommand('autoNumberHeading', {
-				exec: (editor) => {
-					/* Get all header elements and assign classes
-					 * only if needed
-					 */
-					assignClassToList(editor.document.find('h1,h2,h3'));
-				}
-			});
-			
-			// Adding the button to the toolbar
-			editor.ui.addButton( 'structuredheadings', {
-				label: 'Autonumber Heading',
-				command: 'autoNumberHeading',
-				toolbar: 'styles,0'
-			})
-			return editor; // actually, no return is required, but this shuts eslint up
-		}
-	});
-	
-	function assignClassToList(list) {
-		// Iterate over the list of nodes
-		for (var i = 0; i < list.count(); i++) {
-			var h = list.getItem(i);
-			// Add the class if it doesn't have it
-			if(!h.hasClass('autonumber')) {
-				h.addClass('autonumber');
-			}
-		};
-	}
+  CKEDITOR.plugins.add("structuredheadings", {
+    icons: "structuredheadings",
+    init: function (editor) {
+      // list of elements allowed to be numbered
+      var allowedElements = ["h1", "h2", "h3", "h4", "h5", "h6"];
+
+      editor.addContentsCss(this.path + "styles/numbering.css");
```

Comment:
I don't have any knowledge of the reader but I assume we'll have to stick this same css in another file. We can have the main application CSS sheets loaded by the editor, but since the functionality of the plugin depends on this CSS existing I figured it is appropriate to include alongside it. But I guess if this is really just for in-house use it doesn't really matter, and including the styles in the main sheet is the easiest solution.